### ISSUE TITLE
Daily builds: "Find in Files" broken?

### ISSUE THREAD

**Issue by pat-s** [rstudio/rstudio#7667]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.747
    OS Version      : macOS 10.15.6
    R Version       : 4.0.2

### Steps to reproduce the problem

Search for any term in "Find in Files"

### Describe the problem in detail

No results are found, even for the simplest terms. The search result is also returned immediately.

### Describe the behavior you expected

Valid results should show up.

---

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Sorry, but I'm not able to reproduce. Can you provide a reproducible example?

**Comment by pat-s** [comment]
Hi Kevin,

I just found that this happens in only some projects. This gives me headaches.

Already rebooted the machine, same story.
In one affected package, it looks like the search is not even being executed since the results are returned immediately.
Every other search takes at least a second or so.

No filters set, no regex enabled. Just the default "Find in Files" search.

Do you have any idea what I can do here? Or how I can provide a helpful reprex besides recording a GIF?

**Comment by smingerson** [comment]
I've experienced the same, only in package projects as far as I recall, but have not been able to pin it down either. What it seems like to me is that it is not descending into child directories. For example, I will search something, it will immediately return with no results, but if I then select the R/ directory and re-search, I will get results.

**Comment by pat-s** [comment]
Interesting. In the affected package I do not have any usual sub-directories. Also filtering R scripts only does not help.

Could it be that the search index for specific projects might be damaged somehow and needs to flushed first?

**Comment by kevinushey** [comment]
Are you able to share the sources publicly for one of the affected projects?

Does clearing the project state help? You can remove the internal project state with e.g.

```
mv .Rproj.user .Rproj.user.backup
```

and then restart RStudio for that project.

**Comment by pat-s** [comment]
Clearing the project state did not help.

The one in question is [krlmlr/fledge](https://github.com/krlmlr/fledge).

**Comment by melissa-barca** [comment]
Hi @pat-s I am not able to reproduce with the project you provided, can you provide a search phrase that's causing you issues?

**Comment by kevinushey** [comment]
Can you also confirm what you see in the Find in Files dialog when it's opened? E.g.

<img width="416" alt="Screen Shot 2020-08-28 at 1 51 37 PM" src="https://user-images.githubusercontent.com/1976582/91614065-a93b5d80-e935-11ea-8d1b-7ceebd1d9140.png">

One possibility is that the "Search in" path may be wrong if that project was moved and we had cached an older (now invalid) path to the project, but clearing the project settings should've cleared that as well.

**Comment by pat-s** [comment]
Sorry for the late reply. Any search phrase causes the issue. I cannot use it at all in this project. Others are fine though.

Just confirmed again with v1.4.766.

> One possibility is that the "Search in" path may be wrong if that project was moved and we had cached an older (now invalid) path to the project, but clearing the project settings should've cleared that as well.

The project was not moved recently, at least not since I noticed this issue.

---

I found the issue:

<img width="209" alt="image" src="https://user-images.githubusercontent.com/8430564/92007276-04aa8880-ed46-11ea-862e-33915655c97a.png">

Fixing the path solved the issue. Also after a restart the correct path is picked up.
Now this really looks like a "moving issue", however I cannot remember if I really moved the project.

Maybe RStudio could set the search path to the project root by default? Currently it looks like it is not updated when the project is moved or the search path was manually adjusted?

However, I definitely removed the project settings and the search path stayed "incorrect" - maybe that info is helpful :) 

Thanks for all the pointers and help 🤟
